packages/plugins/plugin-table/src/components/Table.tsx

```diff
@@ -2,37 +2,34 @@
 // Copyright 2024 DXOS.org
 //
 
-import React, { useCallback, useRef } from 'react';
+import * as ModalPrimitive from '@radix-ui/react-popper';
+import React, { useCallback, useRef, useState, type MouseEvent } from 'react';
 
 import { type DxGridElement, type DxAxisResize, Grid } from '@dxos/react-ui-grid';
 import { mx } from '@dxos/react-ui-theme';
 
+import { ColumnActionsMenu } from './ColumnActionsMenu';
 import { TableCellEditor } from './TableCellEditor';
 import { useTableModel } from '../hooks';
+import { columnSettingsButtonAttr } from '../table-model';
 import { type TableType } from '../types';
 
-type TableProps = {
-  table: TableType;
-  data: any[];
-};
-
+// NOTE(Zan): These fragments add border to inline-end and block-end of the grid using pseudo-elements.
+// These are offset by 1px to avoid double borders in planks.
 const inlineEndLine =
   '[&>.dx-grid]:relative [&>.dx-grid]:after:absolute [&>.dx-grid]:after:inset-block-0 [&>.dx-grid]:after:-inline-end-px [&>.dx-grid]:after:is-px [&>.dx-grid]:after:bg-separator';
-
 const blockEndLine =
   '[&>.dx-grid]:before:absolute [&>.dx-grid]:before:inset-inline-0 [&>.dx-grid]:before:-block-end-px [&>.dx-grid]:before:bs-px [&>.dx-grid]:before:bg-separator';
 
 const frozen = { frozenRowsStart: 1 };
 
-// NOTE: The table model manages both ephemeral and persistent state.
-// - Ephemeral state (e.g., sorting, filtering, selection) is handled internally.
-// - Persistent state (e.g., column order, widths) is propagated to the parent for storage.
+type TableProps = { table: TableType; data: any[] };
```

Comment:
i think data => rows?

Reply:
also should this move to react-ui-table next?
i guess we should address how existing apps like devtools should handle the transition?